### DISCUSSION TITLE
MonacoWebpackPlugin causing 'TypeError: Cannot set property 'chunkLoading' of undefined'

### DISCUSSION THREAD

**Discussion by mac501pl** [microsoft/monaco-editor#3765] [Q&A]
I'm creating an Electron app using [react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor), [Forge](https://www.electronforge.io/) and Webpack 5. Using MonacoWebpackPlugin in Webpack config causes Forge to throw the following error:
```
$ yarn start
yarn run v1.22.10
$ cross-env NODE_ENV=development electron-forge start
✔ Checking your system
✔ Locating Application
✔ Preparing native dependencies
✔ Compiling Main Process Code
✖ Launch Dev Servers

An unhandled error has occurred inside Forge:
Cannot set property 'chunkLoading' of undefined
TypeError: Cannot set property 'chunkLoading' of undefined
    at WebWorkerTemplatePlugin.apply (C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\webpack\lib\webworker\WebWorkerTemplatePlugin.js:20:40)
    at Compiler.createChildCompiler (C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\@electron-forge\plugin-webpack\node_modules\webpack\lib\Compiler.js:561:12)
    at Compilation.createChildCompiler (C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\@electron-forge\plugin-webpack\node_modules\webpack\lib\Compilation.js:2243:24)
    at C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\monaco-editor-webpack-plugin\out\plugins\AddWorkerEntryPointPlugin.js:17:43
    at AsyncParallelHook.eval [as callAsync] (eval at create (C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:25:1)
    at AsyncParallelHook.lazyCompileHook (C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\tapable\lib\Hook.js:154:20)
    at C:\Users\XXX\Desktop\8BitSimulatorV2\node_modules\@electron-forge\plugin-webpack\node_modules\webpack\lib\Compiler.js:669:20
SimulatorV2\node_modules\@electron-forge\plugin-webpack\node_modules\webpack\lib\Compiler.js:529:11)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Here's a [VERY minimal reproducible example](https://github.com/mac501pl/monaco-issue) (using npm)
And here's a bit less [minimal reproducible example](https://github.com/mac501pl/8080SimV2) (using yarn)

- OS: Windows
- Env: Electron
- Node version: 15.14.0
- monaco-editor-webpack-plugin version: 3.0.1

**Comment by iamaseem** [comment]
Facing same issue with nextjs

**Comment by dgrechka** [comment]
The same for me. the same issue.

**Comment by dgrechka** [comment]
Downgrading back to explicit version `"monaco-editor-webpack-plugin": "3.0.0"` helped, the issue has gone.
That can be a workaround for others.

**Comment by mac501pl** [comment]
Downgrading didn't work for me

**Comment by delagen** [comment]
May be also related to Webpack 5?

**Comment by mac501pl** [comment]
This might very well be the case, apparently Forge doesn't support Webpack 5's definePlugin. Downgrading to Webpack 4 solves the issue.

**Comment by gavinbarron** [comment]
This cropped up for us yesterday when our CI for building a Storybook site started using nodejs 16.20.1 instead of 16.20.0
Frankly I'm a bit stumped as to how to fix this short of forcing our CI to use 16.20.0
